derp/derp_server.go

```diff
@@ -635,6 +661,12 @@ func (c *sclient) sendPkt(dst *sclient, p pkt) error {
 	// this case to keep reader unblocked.
 	s.packetsDropped.Add(1)
 	s.packetsDroppedQueueTail.Add(1)
+	if verboseDropKeys[dstKey] {
+		// Generate a full string including src and dst, so
+		// the limiter kicks in once per src.
+		msg := fmt.Sprintf("head drop %s -> %s", p.src.ShortString(), dstKey.ShortString())
```

Comment:
if you don't want to pay this Sprintf cost all the time (for the cases where the limitedLogf doesn't log), you can use https://godoc.org/tailscale.com/types/logger#ArgWriter as the arg to limitedLogf:

```
c.s.limitedLogf("head drop %s", logger.ArgWriter(func( .... {

}}))
```

Reply:
That's going to rate-limit regardless of what the argument is, whereas I want it to limit on the fully formatted string. AFAICT, to do that I have to preformat the string and pass it in, given our current limiter config.

Given that the impact of doing this will be limited to the customer being debugged, I think I'm okay paying this price for now in exchange for the right debug data.